### DISCUSSION TITLE
Jqxtree is taking around 3-4 seconds to load 500 records which is very slow as it is hampering my app performance.

### DISCUSSION THREAD

**Discussion by ajaykumaretw** [community/community#106577] [Programming Help]
### Select Topic Area

Question

### Body

Jqxtree is taking around 3-4 seconds to load 500 records which is very slow as it is hampering my app performance.

var source = {
datatype: “json”,
datafields: [
{ name: “Rid”, type: “int” },
{ name: “ParentRid”, type: “int” },
{ name: “Name”, type: “string” },
{ name: “Text”, type: “string” }
],
id: “Rid”,
localdata: dataSource,
async: true,
hasThreeStates: false
};
var dataAdapter = new $.jqx.dataAdapter(source);
dataAdapter.dataBind();

var records = dataAdapter.getRecordsHierarchy(“Rid”, “ParentRid”, “items”, [{ name: “Text”, map: “label” }, { name: “Rid”, map: “value” }, { name: “Name”, map: “Name” }]);

$(“#jqxtree”).jqxTree({ checkboxes: true, source: records, width: ‘100%’, height: 220 });

I want to load ‘JqxTree’ control in such a way that other things won’t wait for it to load. Is there any way in the jQuery to load ‘JqxTree’ control in the background?

Or something we can do which won’t affect performance of whole application

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ajaykumaretw ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

If you're asking how to use the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.
